src/main/java/newbank/server/NewBank.java

```diff
@@ -63,19 +72,51 @@ private boolean credentialsAreValid(final String username, final String password
   }
 
   // commands from the NewBank customer are processed in this method
-  public synchronized String processRequest(CustomerID customer, String request) {
-    if (customers.containsKey(customer.getKey())) {
-      switch (request) {
-        case "SHOWMYACCOUNTS":
-          return showMyAccounts(customer);
-        default:
-          return "FAIL";
+  public synchronized String processRequest(CustomerID customerID, String request) {
+    Customer customer = customers.get(customerID.getKey());
+    if (customer == null) {
+      return "FAIL: Customer not found!";
+    }
+
+    String[] args = request.split("\\s+");
+    if (args.length == 0) {
+      return "FAIL: Invalid command!";
+    }
+
+    String command = args[0];
+    switch (command) {
+    case "SHOWMYACCOUNTS":
+      return showMyAccounts(customer);
+
+    case "NEWACCOUNT":
+      if (args.length != 2) {
+        return "FAIL: The proper syntax is: NEWACCOUNT <Name>";
       }
+
+      return newAccount(customer, args[1]);
+
+    default:
+      return "FAIL: Unknown command!";
     }
-    return "FAIL";
   }
 
-  private String showMyAccounts(CustomerID customer) {
-    return (customers.get(customer.getKey())).accountsToString();
+  private String showMyAccounts(Customer customer) {
+    return customer.accountsToString();
+  }
+
+  private String newAccount(Customer customer, String accountName) {
+    try {
```

Comment:
You can simplify using one try statement with multiple catch statements instead of nested try statements...

Kindest Regards,
Jules

Reply:
Ahhh I feel so ashamed as a JavaScript/TypeScript programmer - it's exactly the same there. :D (though, for my defense, we are lazier with exception types!)

Thanks Jules, so clean now!

Reply:
Lol. Np.

Kindest Regards,
Jules